### PR TITLE
[Bug] | [Change]: Homarr (+Script harmonized)

### PR DESCRIPTION
**Pull request by MickLesk**
## Description

Git Pull is not the state of the art in our scripts.
Script updated to latest default. Update with Releases.

## Type of change
Please check the relevant option(s):

- [x] Bug fix (non-breaking change that resolves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [x] Breaking change (a fix or feature that would cause existing functionality to change unexpectedly)
- [ ] New script (a fully functional and thoroughly tested script or set of scripts.)
- [x] Documentation update required (this change requires an update to the documentation)

## Prerequisites
The following efforts must be made for the PR to be considered. Please check when completed:
- [x] Self-review performed (I have reviewed my code, ensuring it follows established patterns and conventions)
- [x] Testing performed (I have tested my changes, ensuring everything works as expected)

## Additional Information (optional)
Provide any additional context or screenshots about the feature or fix here.

## Related Pull Requests / Discussions

If there are other pull requests or discussions related to this change, please link them here:
- Related Issue: https://github.com/community-scripts/ProxmoxVE/issues/119

### CODE REVIEW

File: ct/homarr.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 source <(curl -s https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/misc/build.func)
 # Copyright (c) 2021-2024 tteck
-# Author: tteck (tteckster)
+# Author: MickLesk (Canbiz)
```
Comment (Mellowlynx) [inline ct/homarr.sh:4]: I would leave tteck as the original author and add yourself as a revision author, maybe. 🤔

File: install/homarr-install.sh [hunk 2]
```diff
@@ -1,9 +1,10 @@
 #!/usr/bin/env bash
 
 # Copyright (c) 2021-2024 tteck
-# Author: tteck (tteckster)
```
Comment (Mellowlynx) [inline install/homarr-install.sh:4]: Same

**Review by Mellowlynx — CHANGES_REQUESTED** [review]

File: install/homarr-install.sh [hunk 3]
```diff
@@ -31,12 +31,15 @@ msg_ok "Set up Node.js Repository"
 msg_info "Installing Node.js/Yarn"
 $STD apt-get update
 $STD apt-get install -y nodejs
-$STD npm install -g npm@latest
 $STD npm install -g yarn
 msg_ok "Installed Node.js/Yarn"
 
 msg_info "Installing Homarr (Patience)"
-$STD git clone -b dev https://github.com/ajnart/homarr.git /opt/homarr
+cd /opt
```
Comment (havardthom) [inline install/homarr-install.sh:38]:
```suggestion
```

File: ct/homarr.sh [hunk 4]
```diff
@@ -54,20 +54,38 @@ function default_settings() {
 
 function update_script() {
 header_info
-if [[ ! -d /opt/homarr ]]; then msg_error "No ${APP} Installation Found!"; exit; fi
-msg_info "Updating $APP (Patience)"
-systemctl stop homarr
-cd /opt/homarr
-if ! git pull; then
-  echo "Already up to date."
+if [[ ! -d /opt/adventurelog ]]; then msg_error "No ${APP} Installation Found!"; exit; fi
```
Comment (havardthom) [inline ct/homarr.sh:57]:
```suggestion
if [[ ! -d /opt/homarr]]; then msg_error "No ${APP} Installation Found!"; exit; fi
```

File: ct/homarr.sh [hunk 5]
```diff
@@ -54,20 +54,38 @@ function default_settings() {
...
+if [[ ! -f /opt/${APP}_version.txt ]] || [[ "${RELEASE}" != "$(cat /opt/${APP}_version.txt)" ]]; then
+  msg_info "Stopping Services"
+  systemctl stop homarr
+  msg_ok "Services Stopped"
+
+  msg_info "Updating ${APP} to ${RELEASE}"
+  cp /opt/homarr/.env /opt/.env
+  wget -q "https://github.com/ajnart/homarr/archive/refs/tags/v${RELEASE}.zip"
+  unzip -q v${RELEASE}.zip
+  mv homarr-${RELEASE} /opt/homarr
```
Comment (havardthom) [inline ct/homarr.sh:72]: delete old directory first?

File: ct/homarr.sh [hunk 6]
```diff
@@ -54,20 +54,38 @@ function default_settings() {
...
+  mv /opt/.env /opt/homarr/.env
+  yarn install
+  yarn build
+  yarn db:migrate
+
```
Comment (havardthom) [inline ct/homarr.sh:77]: update `/opt/${APP}_version.txt`

**Review by havardthom — CHANGES_REQUESTED** [review]

File: ct/homarr.sh (hunk 1 above)
Comment (havardthom) [inline ct/homarr.sh:4]:
```suggestion
# Author: tteck (tteckster)
# Co-Author: MickLesk (Canbiz)
```

File: install/homarr-install.sh [hunk 7]
```diff
@@ -1,9 +1,10 @@
 #!/usr/bin/env bash
 
 # Copyright (c) 2021-2024 tteck
-# Author: tteck (tteckster)
+# Author: MickLesk (Canbiz)
```
Comment (havardthom) [inline install/homarr-install.sh:4]:
```suggestion
# Author: tteck (tteckster)
# Co-Author: MickLesk (Canbiz)
```

File: ct/homarr.sh (hunk 5 above)
Comment (havardthom) [reply ct/homarr.sh:72]: but should also doublecheck that there is no data/config there that needs to be migrated

File: install/homarr-install.sh (hunk 3 above)
Comment (MickLesk) [reply install/homarr-install.sh:39]: why?

Comment (havardthom) [reply install/homarr-install.sh:39]: It is not needed

Comment (havardthom) [reply install/homarr-install.sh:39]: There is `cd /opt/homarr` on line 51 before install

File: ct/homarr.sh (hunk 5 above)
Comment (MickLesk) [reply ct/homarr.sh:75]: done

File: ct/homarr.sh [hunk 8]
```diff
@@ -1,7 +1,8 @@
-#!/usr/bin/env bash
+c#!/usr/bin/env bash
```
Comment (havardthom) [inline ct/homarr.sh:1]:
```suggestion
#!/usr/bin/env bash
```

File: install/homarr-install.sh [hunk 9]
```diff
@@ -72,6 +76,7 @@ motd_ssh
 customize
 
 msg_info "Cleaning up"
+rm -rf /opt/v${RELEASE}.zip
```
Comment (havardthom) [inline install/homarr-install.sh:79]:
```suggestion
rm -rf v${RELEASE}.zip
```

File: ct/homarr.sh [hunk 10]
```diff
@@ -54,20 +55,43 @@ function default_settings() {
...
+
+  msg_info "Updating ${APP} to ${RELEASE}"
+  cp /opt/homarr/.env /opt/.env
+  cp -a /opt/homarr/data /opt/
+  rm -rf /opt/homarr
+  wget -q "https://github.com/ajnart/homarr/archive/refs/tags/v${RELEASE}.zip"
+  unzip -q v${RELEASE}.zip
+  mv homarr-${RELEASE} /opt/homarr
+  mv /opt/.env /opt/homarr/.env
+  mv /opt/data /opt/homarr/
```
Comment (havardthom) [inline ct/homarr.sh:77]:
```suggestion
  mv /opt/data /opt/homarr/data
```

**Review by havardthom — APPROVED** [review]

**Review by Mellowlynx — APPROVED** [review]

**Pull request merged**